pandas/tests/internals/test_internals.py

```diff
@@ -1216,7 +1236,24 @@ def test_binop_other(self, op, value, dtype):
         }
 
         if (op, dtype) in invalid:
-            with pytest.raises(TypeError):
+            if op is operator.add:
+                msg = "cannot add DatetimeArray and Timestamp"
+            elif op is operator.mul:
+                msg = (
+                    re.escape(
+                        "ufunc 'multiply' cannot use operands with "
+                        "types dtype('<m8[ns]') and dtype('<m8[ns]')"
+                    )
+                    + "|cannot perform __mul__ with this index type: DatetimeArray"
+                )
+            else:
+                msg = (
+                    re.escape(
+                        f"cannot perform __{op.__name__}__ with this index type: "
+                    )
+                    + "(DatetimeArray|TimedeltaArray)"
+                )
+            with pytest.raises(TypeError, match=msg):
```

Comment:
So in light of the discussion in https://github.com/pandas-dev/pandas/issues/30999. Personally, I find that the above code change makes the test much more complicated and more difficult to read and interpret. 

There is always a trade-off between testing every exact details vs practicality/readabiity/effort to do this, and *personally* for a case like the above, this is not worth it.

Reply:
> This test could benefit from a TON of refactoring, including `pytest` parameterization.

@gfyoung I am looking into refactoring this test from the ground up, what would you put as the  pytest parameterization(s) ?

I hope I can build the test from the ground up, depends on the parameterization.

Reply:
> I am looking into refactoring this test from the ground up, what would you put as the pytest parameterization(s) ?

I might have jumped the gun there in saying we should do `pytest` parameterization.  If there is a way to parameterize so we can remove the logic for the `skip` tests (and put it as part of the decorator), that would be good.

A similar point could be made about the combinations we expect to fail.  If we could somehow parameterize properly, we could easily break that part out as a separate test.